sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -74,6 +74,10 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
     assert(complexData.filter(complexData("a").getItem(0) === 2).count() == 1)
     assert(complexData.filter(complexData("m").getItem("1") === 1).count() == 1)
     assert(complexData.filter(complexData("s").getField("key") === 1).count() == 1)
+
+    // SPARK-24313: access binary keys
```

Comment:
can we create a new test case? I think a correctness bug deserves an individual test case.